azure-pipelines.yml

```diff
@@ -6,93 +6,83 @@
-            version: '5.x'
-            installationPath: $(Agent.ToolsDirectory)/dotnet
-            workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
-        # run build
-        - script: dotnet build --configuration $(buildConfiguration)
-          displayName: dotnet build
-          workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
-        # publish build to artifact
-        - script: dotnet publish --configuration $(buildConfiguration) --output publish
-          displayName: dotnet publish
-          workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
-        
-        - publish: 'FakebookNotifications/FakebookNotifications.WebApi/publish'
-          artifact: app
+    - job: build
+      variables:
+        buildConfiguration: 'Release'
+        directory: 'FakebookNotifications/FakebookNotifications.WebApi'
+      steps:
+      - task: UseDotNet@2
+        displayName: dotnet sdk 5.x
+        inputs:
+          packageType: 'sdk'
+          version: '5.x'
+          installationPath: $(Agent.ToolsDirectory)/dotnet
+          workingDirectory: '$(directory)'
+      # build and publish app
+      - script: dotnet build --configuration $(buildConfiguration)
+        displayName: dotnet build
+        workingDirectory: '$(directory)/'
+      - script: dotnet publish --configuration $(buildConfiguration) --output publish
+        displayName: dotnet publish
+        workingDirectory: '$(directory)'
+      # publish artifact for eventual release
+      - publish: '$(directory)/publish'
+        artifact: app
+
   - stage: SonarCloud
     jobs:
     - job: build
-      # use ubuntu latest
       pool:
         vmImage: ubuntu-latest
-        # use an earlier version of .net because sonar cloud doesn't like .net 5
       variables:
         sdkVersion: 3.1.x
         buildConfiguration: 'Release'
-        #use sonar cloud token 
+        directory: 'FakebookNotifications/FakebookNotifications.WebApi'
+
       steps:
-      - task: UseDotNet@2
-        displayName: dotnet sdk 2.x
+      # use dotnet 2 for sonar compatibility
+      - task: 'UseDotNet@2'
+        displayName: 'dotnet sdk 2.x'
         inputs:
           packageType: 'sdk'
           version: '2.x'
-      - task: SonarCloudPrepare@1
+      # use the Sonar Cloud Token from the available connections
+      - task: 'SonarCloudPrepare@1'
         inputs:
           SonarCloud: 'SonarCloud Token'
           organization: '2011-fakebook-project3'
           scannerMode: 'MSBuild'
           projectKey: '2011-fakebook-project3_notifications'
           extraProperties: |
-                sonar.exclusions=**/lib/**
-                sonar.cs.opencover.reportsPaths = $(Agent.TempDirectory)/**/coverage.opencover.xml
-    # use the latest .net
-      - task: UseDotNet@2
+                'sonar.exclusions=**/lib/**'
+                'sonar.cs.opencover.reportsPaths=$(Agent.TempDirectory)/**/coverage.opencover.xml'
```

Comment:
this is incorrect yaml syntax, the `'` marks are being treated as part of the string. just edit this in the browser-based editor using the 'settings' link above it.